Automattic/woocommerce-payments
# Allow UI Customizations on WCPay, UPE, & Other Checkout Payment Fields

Fixes #3230

* Allows merchants to add UI customizations to their checkout payment fields for WCPay, UPE, and Other (Giropay, Sepa, & Sofort) payment methods. 





#### Changes proposed in this Pull Request

This PR allows merchants to make UI customizations to the checkout payment fields. It follows a similar approach to the approach that was used in [woocommerce-gatewaye-stripe/includes/class-wc-gateway-stripe.php:343](https://github.com/Automattic/woocommerce-gateway-stripe/blob/master/includes/class-wc-gateway-stripe.php#L343).

In order to make UI customizations to the checkout payment fields, the merchant will need to do the following:

<ins>Used</ins>
* **WCPay**: hook (`add_action(...)`) into the following action: `wcpay_payment_fields_wcpay`.
* **UPE**: hook into the following action: `wcpay_payment_fields_upe`.

<ins>Unused</ins> (but might be used in the future)
* **Giropay**: hook into the following action: `wcpay_payment_fields_giropay`.
* **Sepa**: hook into the following action: `wcpay_payment_fields_sepa`.
* **Sofort**: hook into the following action: `wcpay_payment_fields_sofort`.

Note that tests were not added for this minor update. Although, the manual testing instructions (below) should be performed and their expected results should be confirmed.





#### Testing instructions

In order to perform the following tests, please ensure that the following plugin has been installed and activated: `Code Snippets`.



<ins>Preview UI customizations for WCPay</ins>
1. From the admin perspective (`wp-admin`), click on the `WCPay Dev` menu item.
2. Make sure `Enable UPE checkout` is not checked.
3. Go to `Snippets` (requires `Code Snippets` plugin) and click `Add New`.
4. Enter any title and paste the following code snippet in the `Code` section:

```php
add_action('wcpay_payment_fields_wcpay', function ($id) {
  echo "<div style='color: green;'>Thank you for your purchase</div>";
});
```
5. Click `Save Changes`.
6. From the customer perspective, go to the `Shop`, add some product to your cart, and go to `Checkout`.

Note that below the checkout payment fields, now there is some green text that reads: "Thank you for your purchase".

<ins>Preview UI customizations for UPE</ins>
1. From the admin perspective (`wp-admin`), click on the `WCPay Dev` menu item.
2. Make sure `Enable UPE checkout` is checked.
3. Go to `Snippets` (requires `Code Snippets` plugin) and click `Add New`.
4. Enter any title and paste the following code snippet in the `Code` section:

```php
add_action('wcpay_payment_fields_upe', function ($id) {
  echo "<div style='color: purple;'>Thank you for your purchase</div>";
});
```
5. Click `Save Changes`.
6. From the customer perspective, go to the `Shop`, add some product to your cart, and go to `Checkout`.

Note that below the checkout payment fields, now there is some purple text that reads: "Thank you for your purchase".

Finally, it should also be noted that the other payment gateways (Giropay, Sepa, and Sofort) cannot be tested at the moment since they are not being used, outside of UPE. But the `do_action(...)` code was added to the corresponding `payment_fields()` method, in order to future proof them.



-------------------


**Post merge**
